src/main/java/inu/parser/Parser.java

```diff
@@ -0,0 +1,216 @@
+package inu.parser;
+
+import inu.exceptionhandling.EmptyTaskListException;
+import inu.exceptionhandling.EmptyUserInputException;
+import inu.exceptionhandling.ExceptionManager;
+import inu.task.DeadLine;
+import inu.task.Event;
+import inu.task.Task;
+import inu.task.Todo;
+import inu.task.TaskList;
+import inu.commons.Ui;
+import inu.commons.Util;
+
+
+public class Parser {
+
+    protected static final int DECODED_TASK_TYPE = 0;
+
+    protected static final int DECODED_MARK = 1;
+
+    protected static final int DECODED_TASK = 2;
+
+    protected static final int DECODED_DUE_DATE = 3;
+
+    protected static final int DECODED_FROM_DATE = 3;
+
+    protected static final int DECODED_TO_DATE = 4;
+
+    protected static final String T = "T";
+
+    protected static final String D = "D";
+
+    protected static final String E = "E";
```

Comment:
Constant names are exactly the same as the string assigned to them. Think why we are using constants, perhaps give more meaningful names?